src/Microsoft.PowerApps.TestEngine/PowerApps/PowerAppFunctions.cs

```diff
@@ -409,5 +410,36 @@ public async Task<object> GetDebugInfo()
                 throw;
             }
         }
+
+        public async Task<bool> TestEngineReady()
+        {
+            try
+            {
+                // check if ready function exists in the webplayer JSSDK, older versions won't have this new function
+                var checkIfReadyExists = await _testInfraFunctions.RunJavascriptAsync<string>(CheckPowerAppsTestEngineReadyFunction);
+                if (checkIfReadyExists != "undefined")
+                {
+                    var expression = $"PowerAppsTestEngine.testEngineReady()";
+                    return await _testInfraFunctions.RunJavascriptAsync<bool>(expression);
+                }
+
+                // To support webplayer version without ready function 
+                // return true for this without interrupting the test run
+                return true;
```

Comment:
is there a scenario where Ready would be `false` and interrupt the test run?

Reply:
The way this function is setup it will never return false but throw exception (from webplayer app jssdk) in different scenarios